src/main/scala/nl.knaw.dans.easy.deposit/EasyDepositApiApp.scala

```diff
@@ -63,22 +64,30 @@ class EasyDepositApiApp(configuration: Configuration) extends DebugEnhancedLoggi
     val dir = getConfiguredDirectory("deposits.stage-zips")
     logger.info(s"Uploads are staged in $dir")
     if (dir.nonEmpty) {
-      val msg = s"Pending uploads were probably interrupted. See logs lines with 'POST /deposit/{id}/file/{dir_path}'. Please remove their directories from the staging area: $dir"
-      logger.error(msg)
-      throw new FileAlreadyExistsException(msg)
+      // TODO move to Validation and/or new Uploader class for DepositServlet post("/:uuid/file/*")
+      throw new FileAlreadyExistsException(
+        s"Upload staging area [$dir] should be empty unless force shutdown during an upload request. Check logging related to 'POST /deposit/{id}/file/{dir_path}'."
+      )
     }
     dir
   }
-  private val draftsDir = getConfiguredDirectory("deposits.drafts")
+  private val draftsDir: File = getConfiguredDirectory("deposits.drafts")
+  private val provider: FileSystemProvider = uploadStagingDir.fileSystem.provider()
+  Validation.sameMounts(provider, uploadStagingDir, draftsDir)
 
   private val submitter = new Submitter(
     getConfiguredDirectory("deposits.stage-for-submit"),
     getConfiguredDirectory("deposits.submit-to"),
     configuration.properties.getString("deposit.permissions.group"),
   )
 
-  private def getConfiguredDirectory(key: String): File = {
-    val dir = File(configuration.properties.getString(key))
+  @throws[ConfigurationException]("when no existing readable directory is configured")
+  def getConfiguredDirectory(key: String): File = {
+    // TODO move to Validation?
+    val str = Try{configuration.properties.getString(key)}.recoverWith { case e =>
```

Comment:
`getString` returns `null` of the property is not present. `Try` doesn't help you here...
maybe use `Option(...).getOrElse { throw ... }`?